pkg/client/host_orchestrator_client.go

```diff
@@ -284,11 +290,14 @@ func (c *HostOrchestratorServiceImpl) waitForOperation(name string, res any, ret
 	return c.HTTPHelper.NewPostRequest(path, nil).JSONResDoWithRetries(res, retryOpts)
 }
 
-func (c *HostOrchestratorServiceImpl) FetchArtifacts(req *hoapi.FetchArtifactsRequest, creds string) (*hoapi.FetchArtifactsResponse, error) {
+func (c *HostOrchestratorServiceImpl) FetchArtifacts(req *hoapi.FetchArtifactsRequest, creds BuildAPICredential) (*hoapi.FetchArtifactsResponse, error) {
 	var op hoapi.Operation
 	rb := c.HTTPHelper.NewPostRequest("/artifacts", req)
-	if creds != "" {
-		rb.AddHeader(c.BuildAPICredentialsHeader, creds)
+	if creds.AccessToken != "" {
```

Comment:
let's create a helper function down below: `addBuildAPICredsHeaders(rb, creds)` that could be reused by create and fetch methods.